packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js

```diff
@@ -22,105 +49,212 @@ describe('Batch Actions component', () => {
-        deleteButton = wrapper.find(elSelector.deleteButton)
-      })
-
-      it('should display the action buttons', () => {
-        const actionButtons = wrapper.findAll(elSelector.ocButton)
-
-        expect(actionButtons.length).toEqual(3)
-        expect(copyButton.text()).toEqual('Copy')
-        expect(moveButton.text()).toEqual('Move')
-        expect(deleteButton.text()).toEqual('Delete')
-      })
 
-      it('should call "triggerLocationPicker" when copy button is clicked', async () => {
-        await copyButton.vm.$emit('click')
-
-        expect(spyTriggerLocationPicker).toHaveBeenCalledWith('copy')
-      })
-
-      it('should call "triggerLocationPicker" when move button is clicked', async () => {
-        await moveButton.vm.$emit('click')
+        if (page === 'files-shared-with-me') {
+          let wrapper
+          let actionButtons
+          let acceptButton
+          let declineButton
+          const spyAcceptShares = jest
+            .spyOn(BatchActions.methods, 'acceptShares')
+            .mockImplementation()
+          const spyDeclineShares = jest
+            .spyOn(BatchActions.methods, 'declineShares')
+            .mockImplementation()
+
+          it('should display accept and decline buttons if share status is pending', () => {
+            const store = createStore({
+              selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.pending })
+            })
+            wrapper = createWrapper({ ...options, store })
+            actionButtons = wrapper.findAll(elSelector.ocButton)
+            acceptButton = wrapper.find(elSelector.acceptButton)
+            declineButton = wrapper.find(elSelector.declineButton)
+
+            expect(actionButtons.length).toEqual(2)
+            expect(acceptButton.exists()).toBeTruthy()
+            expect(acceptButton.text()).toContain('Accept')
+            expect(declineButton.exists()).toBeTruthy()
+            expect(declineButton.text()).toContain('Decline')
+          })
+          describe('when share status is declined', () => {
+            beforeEach(() => {
+              const store = createStore({
+                selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.declined })
+              })
+              wrapper = createWrapper({ ...options, store })
+              actionButtons = wrapper.findAll(elSelector.ocButton)
+              acceptButton = wrapper.find(elSelector.acceptButton)
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display accept button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(acceptButton.exists()).toBeTruthy()
+              expect(acceptButton.text()).toContain('Accept')
+              expect(declineButton.exists()).toBeFalsy()
+            })
+            it('should call "acceptShares" when accpet button is clicked', async () => {
+              await acceptButton.vm.$emit('click')
```

Comment:
can you use trigger here and other usages?
Directly emitted *click* on the button does not verify whether the call is initiated after the button click.

Reply:
I have stubbed `ODS` components and used `shallowMount` as per the comments I received previously ([see](https://github.com/owncloud/web/pull/5282#discussion_r653584960))
click cannot be triggered as `.trigger('click')` on stubbed components so have used `emit`

Would you want me to use `mount`?

Reply:
Imo mount should be used while necessary. For test which involves see emitting should use mount to mount components